nucliadb_utils/nucliadb_utils/concurrency.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import asyncio
+from collections.abc import Coroutine
+from typing import Any, Optional
+
+from nucliadb_utils import logger
+
+
+class ConcurrentRunner:
+    """
+    Runs a list of coroutines concurrently, with a maximum number of tasks running.
+    Returns the results of the coroutines in the order they were scheduled.
+    """
+
+    def __init__(self, max_tasks: Optional[asyncio.Semaphore] = None):
```

Comment:
NIT: I'd prefer an `int` here to hide the `asyncio.Semaphore` implementation detail

edit: you're making this in `run_concurrently`, so feel free to ignore this